### PR TITLE
smoke test for fed keyless

### PR DESCRIPTION
**Pull request by zjma**
## Type of Change
- [x] Tests

## Which Components or Systems Does This Change Impact?
n/a

## How Has This Been Tested?
n/a

## Key Areas to Review
function `fedrate_keyless_scenario`.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: types/src/keyless/mod.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ use std::{
 
 mod bn254_circom;
 mod circuit_constants;
-mod circuit_testcases;
+pub mod circuit_testcases;
```
Comment (alinush) [inline types/src/keyless/mod.rs:26]: you shouldn't need to do this

**Review by alinush — COMMENTED** [review]
I think you're exposing `circuit_testcases` unnecessarily. Can you fix that?

File: types/src/keyless/mod.rs (hunk 1 above)
Comment (zjma) [reply types/src/keyless/mod.rs]: fixed!

File: testsuite/smoke-test/src/keyless.rs [hunk 2]
```diff
@@ -225,6 +228,111 @@ async fn test_keyless_groth16_verifies() {
...
+        let e = utf8(b"{}");
+        let n = utf8(b"{}");
+        let jwk = jwks::new_rsa_jwk(kid, alg, e, n);
+        let patch_1 = jwks::new_patch_upsert_jwk(iss, jwk);
+        let patches = vector[patch_0, patch_1]; // clear all, then add 1 jwk.
+        jwks::patch_federated_jwks(account, patches);
+    }}
+}}
+    "#,
+            SAMPLE_TEST_ISS_VALUE, SAMPLE_JWK.kid, SAMPLE_JWK.alg, SAMPLE_JWK.e, SAMPLE_JWK.n
```
[the remark this answers is not in the document]
Comment (zjma) [reply testsuite/smoke-test/src/keyless.rs]: fixed!

File: testsuite/smoke-test/src/keyless.rs [hunk 3]
```diff
@@ -225,6 +227,116 @@ async fn test_keyless_groth16_verifies() {
...
+    fedrate_keyless_scenario(true, false, false).await
+}
+
+#[tokio::test]
+async fn federated_keyless_should_work_otherwise() {
+    fedrate_keyless_scenario(true, true, true).await
+}
+
+/// Config the chain, run a federated keyless txn, and assert txn result.
+async fn fedrate_keyless_scenario(
```
Comment (heliuchuan) [inline testsuite/smoke-test/src/keyless.rs:246]: fedrate_keyless_scenario - spelling

Comment (zjma) [reply testsuite/smoke-test/src/keyless.rs]: fixed

**Review by heliuchuan — APPROVED** [review]

**Review by alinush — APPROVED** [review]

**Pull request merged**